src/bootupd.rs

```diff
@@ -680,25 +633,103 @@ pub(crate) fn client_run_migrate_static_grub_config() -> Result<()> {
             println!("GRUB config symlink successfully replaced with the current config");
 
             // Remove the now unused symlink (optional cleanup, ignore any failures)
-            _ = dirfd.remove_file(&stripped_config);
+            _ = dirfd.remove_file(stripped_config);
         }
     };
 
     println!("Setting 'sysroot.bootloader' to 'none' in ostree repo config...");
+    set_ostree_bootloader("none")?;
+
+    println!("Static GRUB config migration completed successfully");
+    Ok(())
+}
+
+#[context("Get sysroot.bootloader in ostree repo config")]
+pub(crate) fn get_ostree_bootloader() -> Result<Option<String>> {
+    let mut cmd = std::process::Command::new("ostree");
+    let result = cmd
+        .args([
+            "config",
+            "--repo=/sysroot/ostree/repo",
+            "get",
+            "sysroot.bootloader",
+        ])
+        .output()
+        .context("Querying ostree sysroot.bootloader")?;
+    if !result.status.success() {
+        // ostree will exit with a non zero return code if the key does not exists
+        println!("ostree repo 'sysroot.bootloader' config option is not set yet");
+        return Ok(None);
+    } else {
+        let res = String::from_utf8(result.stdout)
+            .with_context(|| "decoding as UTF-8 output of ostree command")?;
+        let bootloader = res.trim_end().to_string();
+        return Ok(Some(bootloader));
+    }
+}
+
+pub(crate) fn set_ostree_bootloader(bootloader: &str) -> Result<()> {
     let status = std::process::Command::new("ostree")
         .args([
             "config",
             "--repo=/sysroot/ostree/repo",
             "set",
             "sysroot.bootloader",
-            "none",
+            bootloader,
         ])
         .status()?;
     if !status.success() {
-        anyhow::bail!("Failed to set 'sysroot.bootloader' to 'none' in ostree repo config");
+        anyhow::bail!("Failed to set 'sysroot.bootloader' to '{bootloader}' in ostree repo config");
     }
+    Ok(())
+}
+
+/// Writes a stripped GRUB config to `stripped_config_name`, removing lines between
+/// `### BEGIN /etc/grub.d/15_ostree ###` and `### END /etc/grub.d/15_ostree ###`.
+fn strip_grub_config_file(
+    current_config_path: &Path,
```

Comment:
I think it'd be cleaner if this was:
```
     current_config_path: impl Read
```
or so. That way everything in the function would consistently operate on `dirfd` which lends itself more easily to unit testing.

(The callers would take care of opening the file, this line would move there)